### PR TITLE
feat: Issues section (replaces Feedback going forward)

### PR DESCRIPTION
**Pull request by peterdrier**
## Summary

Implements the Issues section per spec [`docs/superpowers/specs/2026-04-29-issues-section-design.md`](https://github.com/peterdrier/Humans/blob/feat/issues-section/docs/superpowers/specs/2026-04-29-issues-section-design.md) (spec PR peterdrier/Humans#358).

- New top-level `/Issues` URL with three views (Index / Detail / New) matching the Issue tracker UI kit in the design bundle.
- Section-tagged + role-routed queues — TicketAdmin sees Tickets, CampAdmin sees Camps + CityPlanning, etc. Admin sees everything.
- Six-state lifecycle (Triage / Open / InProgress / Resolved / WontFix / Duplicate) with reporter-comment auto-reopen on terminal issues.
- Threaded comments + inline audit events (status change, assignee change, section change, GitHub link).
- Floating widget on every page now submits to `/Issues` (old `/Feedback` page remains for backlog drainage; deletion is a follow-up PR per spec section 13).
- New `/api/issues/*` surface key-authed via `ISSUES_API_KEY` for LLM agents.
- 23 service-layer unit tests, 12 API controller tests, 24 URL inference tests, plus the existing 5 domain transition tests — all passing.

## What changed at a glance

- **Domain**: 2 enums (`IssueCategory`, `IssueStatus` + `IsTerminal`), 2 entities (`Issue`, `IssueComment` — cross-domain navs `[Obsolete]`-marked per design-rules §6c), 4 new `AuditAction` values, 3 new `NotificationSource` values, 1 new `GdprExportSection`, `IssueSectionRouting` constants.
- **Application**: `IIssuesService` + DTOs, `IIssuesRepository`, `IssuesService` impl with state machine + cross-domain nav stitching + audit logging + notification fan-out + GDPR contributor.
- **Infrastructure**: 2 EF configurations + AddIssues migration (8 indexes, FK delete behaviors per spec) + IssuesRepository (`Attach` + `EntityState.Modified` per codebase convention).
- **Web**: `IssuesController` (resource-based handler-gated mutations), `IssuesApiController` + `IssuesApiKeyAuthFilter`, view models, three Razor views, `IssuesWidgetViewComponent` (cuts over the floating widget), `IssueSectionInference` URL helper, NavBadges integration, `_Layout` + `_AdminLayout` swap, Admin/Configuration `ISSUES_API_KEY` indicator.
- **Email**: `IEmailService.SendIssueCommentAsync` + impls in Smtp/Outbox/Stub services + `RenderIssueComment` in `EmailRenderer` + 6 locales (en/es/de/fr/it/ca).
- **Localization**: 60+ `Issue_*` strings across 6 locales.
- **Docs**: `docs/sections/Issues.md` (section invariant doc), `docs/features/28-issues-system.md` (feature spec).

## Test plan

- [ ] Submit issue from floating widget on /Camps, /Shifts, /Tickets — confirm `Section` is inferred correctly from `PageUrl`
- [ ] Submit issue from /Issues/New with no PageUrl — Section dropdown picks the section
- [ ] As `TicketAdmin` (no Admin), confirm only Tickets-section + own reports show on /Issues
- [ ] As `CampAdmin`, confirm Camps + CityPlanning issues are visible
- [ ] As reporter, confirm only own reports show
- [ ] Change status to Resolved → reporter notification fires
- [ ] Reporter comments on Resolved → status flips to Open, audit event appears in thread
- [ ] Assign issue → assignee notification fires
- [ ] Self-assign → no notification (silent)
- [ ] Re-route Section → new role-holders see it; old role-holders don't
- [ ] Old /Feedback page still reachable; existing reports still usable
- [ ] /api/issues rejects without ISSUES_API_KEY (503 if unset; 401 if mismatched); accepts with matching key
- [ ] EF migration reviewer pass
- [ ] Reporter+assignee both notified on status change (excluding actor)

## Out of scope (per spec)

- Migration of old Feedback reports — parallel-run + drain + delete in a future PR.
- Deletion of old `/Feedback` UI, `FeedbackController`, `FeedbackService`, `FeedbackAdmin` role, `FEEDBACK_API_KEY`, the `feedback_*` tables, the `/triage` Claude Code skill — separate follow-up PR when `feedback_reports` has zero non-terminal rows.
- Priority field, tags/labels, watchers, related-issue links, internal-only comments, time tracking, auto-close on inactivity, email submission.

## Notes

- Per CLAUDE.md memory rule: NO startup guards. Missing `ISSUES_API_KEY` → 503 at request time, never aborts startup.
- Per design-rules §6c: cross-domain nav properties (`Issue.Reporter`, `Issue.Assignee`, `Issue.ResolvedByUser`, `IssueComment.SenderUser`) are `[Obsolete]`-marked. The service stitches them via `IUserService.GetByIdsAsync`. Repository never `.Include()`s a cross-domain nav.
- Per coding-rules: claims-first role check in `IssuesController.CanHandle` and `NavBadgesViewComponent`. No `UserManager.GetRolesAsync` regression.
- One follow-up flagged: per-user `IssuesBadge(userId)` cache has no explicit invalidation hook (relies on 2-min TTL). Same pattern as the pre-existing `VotingBadge`. Easy fix in a follow-up PR if staleness shows up in practice.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: src/Humans.Web/Helpers/IssueSectionInference.cs [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    /// inferred from a path's first segment, or null if no match.
+    /// Examples: "/Camps/123" returns "Camps"; "/Tickets" returns "Tickets"; "/" returns null.
+    /// </summary>
+    public static string? FromPath(string? path)
+    {
+        if (string.IsNullOrWhiteSpace(path)) return null;
+        if (!Uri.TryCreate(path, UriKind.RelativeOrAbsolute, out var uri))
+            return null;
+
+        var p = uri.IsAbsoluteUri ? uri.AbsolutePath : path;
```
[the remark this answers is not in the document]
Comment (peterdrier) [reply src/Humans.Web/Helpers/IssueSectionInference.cs]: Addressed in 23082285: stripped query strings and fragments from relative paths in `IssueSectionInference.FromPath` so `/Tickets?tab=open` correctly routes to the Tickets section. Added 4 test cases covering query-string and fragment edge cases.

File: src/Humans.Web/Models/IssueViewModels.cs [hunk 2]
```diff
@@ -0,0 +1,215 @@
...
+
+    /// <summary>For "comment" rows — true when the comment author is the issue reporter.</summary>
+    public bool ActorIsReporter { get; set; }
+}
+
+public class PostIssueCommentModel
+{
+    [Required]
+    [StringLength(10_000)]
+    public string Content { get; set; } = string.Empty;
```
[the remark this answers is not in the document]
Comment (peterdrier) [reply src/Humans.Web/Models/IssueViewModels.cs]: Already addressed in f387de1c: `PostIssueCommentModel.Content` capped to 5000 to match `issue_comments.Content` column; composer textarea `maxlength` aligned.

File: src/Humans.Web/Models/IssueViewModels.cs [hunk 3]
```diff
@@ -0,0 +1,215 @@
...
+/// Title + Description + Category are required; everything else is optional.
+/// </summary>
+public class SubmitIssueViewModel
+{
+    [Required]
+    [StringLength(500)]
+    public string Title { get; set; } = string.Empty;
+
+    [Required]
+    [StringLength(10_000)]
```
[the remark this answers is not in the document]
Comment (peterdrier) [reply src/Humans.Web/Models/IssueViewModels.cs]: Already addressed in f387de1c (simplify pass): `SubmitIssueViewModel.Title` capped to 200, `Description` to 5000, `ApiCreateIssueModel` aligned, HTML `maxlength` attrs in New/Widget/Detail views aligned to match.

File: src/Humans.Application/Services/Issues/IssuesService.cs [hunk 4]
```diff
@@ -0,0 +1,679 @@
+using Microsoft.AspNetCore.Http;
+using Microsoft.Extensions.Hosting;
```
[the remark this answers is not in the document]
Comment (peterdrier) [reply src/Humans.Application/Services/Issues/IssuesService.cs:2]: Acknowledged but **deferred to a separate cross-section refactor**. `IFormFile` + `IHostEnvironment` + `Directory.CreateDirectory` is the established pattern in `FeedbackService.SubmitFeedbackAsync` (lines 90, 110-138 of the same file), and the Issues plan explicitly directed mirroring it. Refactoring just IssuesService to use `IFileStorage` while leaving FeedbackService violating §1 would create asymmetric architecture. Both should be fixed together — happy to open a follow-up PR for that. The new `IssuesArchitectureTests` (added in 23082285) intentionally mirrors `FeedbackArchitectureTests` and does not assert against this until the cross-section fix lands.

File: src/Humans.Application/Services/Issues/IssuesService.cs [hunk 5]
```diff
@@ -0,0 +1,679 @@
...
+            await using var stream = new FileStream(absolutePath, FileMode.Create);
+            await screenshot.CopyToAsync(stream, ct);
+
+            issue.ScreenshotFileName = screenshot.FileName;
+            issue.ScreenshotStoragePath = relativePath.Replace('\\', '/');
+            issue.ScreenshotContentType = screenshot.ContentType;
+        }
+
+        await _repo.AddIssueAsync(issue, ct);
+        _navBadge.Invalidate();
```
[the remark this answers is not in the document]
Comment (peterdrier) [reply src/Humans.Application/Services/Issues/IssuesService.cs:156]: Doc fixed in 23082285 — the `IssueSubmitted` notification was never in the spec and was overstated in the section invariant doc. Removed both the trigger claim and the `NotificationSource.IssueSubmitted` reference from the cross-section dependencies list. Handlers see new items via the nav-badge actionable count refresh (which IS wired). If we later decide we need a dedicated submit-time fan-out, that's a small additive change.

File: src/Humans.Application/Services/Issues/IssuesService.cs [hunk 6]
```diff
@@ -0,0 +1,679 @@
...
+
+        issue.Section = newSection;
+        issue.UpdatedAt = _clock.GetCurrentInstant();
+        await _repo.SaveTrackedIssueAsync(issue, ct);
+
+        await LogAuditAsync(
+            AuditAction.IssueSectionChanged, issueId, actorUserId,
+            $"Issue {issueId} section: {oldSection} → {nextSection}");
+        _navBadge.Invalidate();
+    }
```
[the remark this answers is not in the document]
Comment (peterdrier) [reply src/Humans.Application/Services/Issues/IssuesService.cs:408]: Fixed in 23082285. `UpdateSectionAsync` now throws `InvalidOperationException` when called on a terminal issue, enforcing the "Section is editable in any non-terminal state" invariant.

File: docs/sections/Issues.md [hunk 7]
```diff
@@ -0,0 +1,186 @@
...
+
+- Every issue is linked to the human who submitted it (`ReporterUserId` is required).
+- Status flows Triage → Open → InProgress → Resolved/WontFix/Duplicate. Transitioning out of a terminal status clears `ResolvedAt` and `ResolvedByUserId`.
+- A reporter posting a comment on a terminal issue **auto-reopens** it to `Open` (audit-logged as `AuditAction.IssueStatusChanged` with actor = the reporter).
+- A handler may post a comment and atomically mark the issue resolved in the same request ("Comment & mark resolved"). The status change is audit-logged after the comment is persisted.
+- Visibility: a regular human sees only the issues they reported. A section role-holder sees all issues whose `Section` maps to one of their roles. Admin sees every issue.
+- Mutation: only handlers (Admin or section role-holders) may change status, assignee, section, or GitHub link, or post a comment as a non-reporter. The reporter may post a comment but cannot change other fields.
+- `Section` is editable in any non-terminal state (handlers may re-route at any time before the issue closes).
+- Screenshots are validated for allowed file types (JPEG, PNG, WebP) and a max size of 5 MB before storage.
```
[the remark this answers is not in the document]
Comment (peterdrier) [reply docs/sections/Issues.md]: Fixed in 23082285 — doc says 10 MB now to match `MaxScreenshotBytes = 10 * 1024 * 1024`.

File: docs/sections/Issues.md [hunk 8]
```diff
@@ -0,0 +1,186 @@
...
+- **Cross-domain navs `[Obsolete]`-marked:** `Issue.Reporter`, `.Assignee`, `.ResolvedByUser`, `IssueComment.SenderUser`. The repository does not `.Include()` them; `IssuesService.StitchCrossDomainNavsAsync` resolves display data in memory from `IUserService` (design-rules §6b). EF still needs the nav refs to wire the DB-level FK + cascade behavior — those references are inside `#pragma warning disable CS0618` blocks in the EF configurations.
+- **Cross-section calls** — the public interfaces this section consumes: `IUserService`, `IUserEmailService`, `IRoleAssignmentService`, `IEmailService`, `INotificationService`, `IAuditLogService`, `INavBadgeCacheInvalidator`.
+- **Nav-badge cache invalidation** routes through `INavBadgeCacheInvalidator` instead of `IMemoryCache` directly.
+- **Architecture test** — `tests/Humans.Application.Tests/Architecture/IssuesArchitectureTests.cs` pins the shape (no EF imports in `IssuesService`, repository is the only `DbContext` consumer for the `issues` / `issue_comments` tables).
```
[the remark this answers is not in the document]
Comment (peterdrier) [reply docs/sections/Issues.md:179]: Created in 23082285: `tests/Humans.Application.Tests/Architecture/IssuesArchitectureTests.cs`, mirroring `FeedbackArchitectureTests`. 9 test methods pinning namespace, no DbContext / IMemoryCache / Stores in the constructor, repository + nav-badge invalidator + cross-section service dependencies, and IIssuesRepository namespace + IssuesRepository sealed implementation.

File: src/Humans.Web/Controllers/IssuesApiController.cs [hunk 9]
```diff
@@ -0,0 +1,305 @@
...
+        AssigneeUserId = i.AssigneeUserId,
+        AssigneeName = i.Assignee?.DisplayName,
+        i.GitHubIssueNumber,
+        i.DueDate,
+        ScreenshotUrl = i.ScreenshotStoragePath is not null ? $"/{i.ScreenshotStoragePath}" : null,
+        CreatedAt = i.CreatedAt.ToDateTimeUtc(),
+        UpdatedAt = i.UpdatedAt.ToDateTimeUtc(),
+        ResolvedAt = i.ResolvedAt?.ToDateTimeUtc(),
+        CommentCount = i.Comments.Count
+    };
```
[the remark this answers is not in the document]
Comment (peterdrier) [reply src/Humans.Web/Controllers/IssuesApiController.cs:260]: Acknowledged. **Deferring** to a follow-up PR. Converting to typed `IssueApiListItem` / `IssueApiDetail` / `IssueApiCommentItem` records is the right move for stability, but it's a moderate-touch change that affects the integration tests and would expand this PR's scope. The current property names are stable (they directly mirror the entity field names), and `MapList` was just hardened with null-safe Reporter access in f387de1c. Will open a separate PR for the typed-DTO conversion against `/api/issues` and `/api/feedback` together.

File: src/Humans.Infrastructure/Repositories/Issues/IssuesRepository.cs [hunk 10]
```diff
@@ -0,0 +1,151 @@
...
+            .AsNoTracking()
+            .Include(i => i.Comments.OrderBy(c => c.CreatedAt))
+            .FirstOrDefaultAsync(i => i.Id == id, ct);
+    }
+
+    public async Task<Issue?> FindForMutationAsync(Guid id, CancellationToken ct = default)
+    {
+        await using var db = await _factory.CreateDbContextAsync(ct);
+        return await db.Issues
+            .Include(i => i.Comments)
```
[the remark this answers is not in the document]
Comment (peterdrier) [reply src/Humans.Infrastructure/Repositories/Issues/IssuesRepository.cs:44]: Acknowledged but **not changing in this PR**. At 500-user scale with `AsNoTracking()` on read paths and a fresh DbContext per call, the cost of the extra `Include(Comments)` on the 4 scalar mutation paths is negligible. The semantics-vs-perf tradeoff is real but minor; splitting into two methods adds an interface change. If this becomes a hot path it's a one-line repo addition.

File: src/Humans.Application/Services/Issues/IssuesService.cs [hunk 11]
```diff
@@ -0,0 +1,685 @@
...
+        var issue = await _repo.FindForMutationAsync(issueId, ct)
+            ?? throw new InvalidOperationException($"Issue {issueId} not found");
+
+        if (string.Equals(issue.Section, newSection, StringComparison.Ordinal)) return;
+
+        if (issue.Status.IsTerminal())
+        {
+            throw new InvalidOperationException(
+                $"Cannot change section on a terminal issue (status: {issue.Status}).");
+        }
```
[the remark this answers is not in the document]
Comment (peterdrier) [reply src/Humans.Application/Services/Issues/IssuesService.cs:395]: Fixed in 1d73f5e4. `IssuesController.UpdateSection` now `SetError(ex.Message)` so a handler re-sectioning a terminal issue sees the actual reason instead of a silent 404. `IssuesApiController.UpdateSection` distinguishes "not found" (404) from state-machine violation (422 UnprocessableEntity) via a message-based filter. Used the codebase-standard catch-and-surface pattern (mirrors CampAdminController) instead of a new exception type — keeps the InvalidOperationException convention consistent across sections.

File: src/Humans.Application/Interfaces/Repositories/IIssuesRepository.cs [hunk 12]
```diff
@@ -0,0 +1,37 @@
...
+    /// <summary>Returns issue with comments (.Include) loaded; cross-domain navs are NOT included.</summary>
+    Task<Issue?> FindForMutationAsync(Guid id, CancellationToken ct = default);
+
+    Task<IReadOnlyList<Issue>> GetListAsync(
+        IssueListFilter filter,
+        IReadOnlySet<string>? sectionFilter,
+        Guid? reporterFallback,
+        CancellationToken ct = default);
+
+    Task<IReadOnlyList<IssueComment>> GetCommentsAsync(Guid issueId, CancellationToken ct = default);
```
[the remark this answers is not in the document]
Comment (peterdrier) [reply src/Humans.Application/Interfaces/Repositories/IIssuesRepository.cs]: Removed in 1d73f5e4 — `GetCommentsAsync` deleted from both `IIssuesRepository` and `IssuesRepository`. Comments are loaded via `GetByIdAsync` (`.Include(i => i.Comments)`) on every read path.

File: src/Humans.Application/Services/Issues/IssuesService.cs [hunk 13]
```diff
@@ -0,0 +1,685 @@
...
+            var relativePath = Path.Combine("uploads", "issues", issueId.ToString(), fileName);
+            var absolutePath = Path.Combine(_env.ContentRootPath, "wwwroot", relativePath);
+
+            Directory.CreateDirectory(Path.GetDirectoryName(absolutePath)!);
+
+            await using var stream = new FileStream(absolutePath, FileMode.Create);
+            await screenshot.CopyToAsync(stream, ct);
+
+            issue.ScreenshotFileName = screenshot.FileName;
+            issue.ScreenshotStoragePath = relativePath.Replace('\\', '/');
```
[the remark this answers is not in the document]
Comment (peterdrier) [reply src/Humans.Application/Services/Issues/IssuesService.cs:151]: The premise ("DB cascade rule means deleting a user deletes their issues rows") was wrong, and the fix below is **not what we want**.

The codebase doesn't actually delete users — `IAccountDeletionService.PurgeAsync` and `AnonymizeExpiredAccountAsync` anonymize the User row in place (display name + email replaced with sentinels, UserEmail rows removed). The User row itself stays. So FKs from issues to the user remain valid; issues persist with their FKs pointing at the (now-anonymized) reporter.

The original Cascade FK was a footgun that didn't fire under normal operation but would have silently wiped issue history if anyone called `db.Users.Remove(user)` directly. **Fixed in 1d73f5e4** — `Issue.Reporter` FK is now `Restrict`. The DB will reject a stray Remove rather than cascade-delete every issue. Section invariant doc updated to describe the actual behavior (anonymize-in-place, FKs intact).

No screenshot cleanup added: issues survive reporter anonymization, so there's nothing to clean up. `ScreenshotStoragePath` not added to GDPR export — this is the user's own reported data, accessible to them via the same /Issues UI; the export already captures the title/description/comments which is what GDPR requires.

File: src/Humans.Infrastructure/Repositories/Issues/IssuesRepository.cs [hunk 14]
```diff
@@ -0,0 +1,151 @@
...
+            .FirstOrDefaultAsync(i => i.Id == id, ct);
```
[the remark this answers is not in the document]
Comment (peterdrier) [reply src/Humans.Infrastructure/Repositories/Issues/IssuesRepository.cs:45]: Acknowledged but **not changing**. Same finding as the prior comment on `FindForMutationAsync` line 43 (already replied) — at 500-user scale and `AsNoTracking()`-free mutation paths, the extra `Include(Comments)` overhead is negligible. Splitting into two repository methods adds an interface surface I'd rather not introduce until there's actual hot-path evidence.

**Comment by peterdrier** [comment]
Rebasing onto main and regenerating the EF migration; will reopen shortly.
